### PR TITLE
misc osd leaks

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/tools/ceph_objectstore_tool.cc [hunk 1]
```diff
@@ -2352,11 +2369,12 @@ int main(int argc, char **argv)
 
     if (!dry_run) {
       superblock.compat_features.incompat.insert(CEPH_OSD_FEATURE_INCOMPAT_SHARDS);
+      ObjectStore::Sequencer osr(__func__);
```
Comment (dzafman) [inline src/tools/ceph_objectstore_tool.cc]: You shouldn't need this one, since you added an osr in main() already.

**Comment by dzafman** [comment]
Minor nit..otherwise :+1:

**Comment by dachary** [comment]
<pre>
./test/osd/osd-config.sh:99: TEST_config_track:  grep 'is not > osd_pg_epoch_persisted_max_stale' testdir/osd-config/osd.0.log
2015-09-15 05:58:00.515475 7f5b59515700  0 log_channel(cluster) log [WRN] : osd_map_cache_size (200) is not > osd_pg_epoch_persisted_max_stale (200)
2015-09-15 05:58:00.969718 7f5b59515700  0 log_channel(cluster) log [WRN] : osd_map_cache_size (200) is not > osd_pg_epoch_persisted_max_stale (200)
./test/osd/osd-config.sh:99: TEST_config_track:  return 1
./test/osd/osd-config.sh:33: run:  return 1
</pre>
https://github.com/ceph/ceph/blob/master/src/osd/OSD.cc#L8596

**Comment by dachary** [comment]
This is because of https://github.com/ceph/ceph/commit/1ff51a299d20dd73a95ba55fdec57498b8c71c13

File: src/os/FileStore.cc [hunk 2]
```diff
@@ -1933,16 +1932,16 @@ int FileStore::queue_transactions(Sequencer *posr, list<Transaction*> &tls,
...
     osr = static_cast<OpSequencer *>(posr->p.get());
-    dout(5) << "queue_transactions existing " << *osr << "/" << osr->parent << dendl; //<< " w/ q " << osr->q << dendl;
+    dout(5) << "queue_transactions existing " << osr << " " << *osr << dendl;
   } else {
     osr = new OpSequencer;
+    osr->set_cct(g_ceph_context);
     osr->parent = posr;
     posr->p = osr;
-    dout(5) << "queue_transactions new " << *osr << "/" << osr->parent << dendl;
+    dout(5) << "queue_transactions new " << osr << " " << *osr << dendl;
```
Comment (tchaikov) [inline src/os/FileStore.cc]: oh, right! the instrusive_ptr's ` intrusive_ptr & operator=(T * r)` does increase the refcount. and `posr` is the only owner of this `OpSequencer`. so should have `Sequencer_impl` start with refcount = 0.

**Comment by tchaikov** [comment]
lgtm also.

**Pull request merged**